From 1db3fb75bff857ac5309a73e9c271f9747f58c35 Mon Sep 17 00:00:00 2001
From: "hjnim.kim" <hjnim.kim@samsung.com>
Date: Wed, 13 Nov 2013 16:23:12 +0900
Subject: [PATCH] JQM Checkboxradio event, style has been synchronized

JQM Checkboxradio event, style has been synchronized
vmouse event, ui-btn-down

Change-Id: Ia9f11ac28dfe53a07f004a684df9954095fa5de2
Signed-off-by: hyunjung kim <hjnim.kim@samsung.com>
---
 .../js/jquery.mobile.buttonMarkup.js               |    6 ++++++
 .../js/widgets/forms/checkboxradio.js              |    4 +++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/libs/js/jquery-mobile-1.2.0/js/jquery.mobile.buttonMarkup.js b/libs/js/jquery-mobile-1.2.0/js/jquery.mobile.buttonMarkup.js
index 7e1bb1f..e7925f5 100644
--- a/libs/js/jquery-mobile-1.2.0/js/jquery.mobile.buttonMarkup.js
+++ b/libs/js/jquery-mobile-1.2.0/js/jquery.mobile.buttonMarkup.js
@@ -281,6 +281,12 @@ var attachEvents = function() {
 						$btn.removeClass( "ui-btn-up-" + theme ).addClass( "ui-btn-down-" + theme );
 					}
 				} else if ( evt === "vmousecancel" || evt === "vmouseup" || evt === "touchend" || evt === "touchcancel" ) {
+					if ( hov ) {
+						clearTimeout( hov );
+					}
+					if ( foc ) {
+						clearTimeout( foc );
+					}
 					$btn.removeClass( "ui-btn-down-" + theme ).addClass( "ui-btn-up-" + theme );
 				} else if ( evt === "vmouseover" || evt === "focus" ) {
 					if ( isTouchEvent ) {
diff --git a/libs/js/jquery-mobile-1.2.0/js/widgets/forms/checkboxradio.js b/libs/js/jquery-mobile-1.2.0/js/widgets/forms/checkboxradio.js
index a8f0082..f9a4b98 100644
--- a/libs/js/jquery-mobile-1.2.0/js/widgets/forms/checkboxradio.js
+++ b/libs/js/jquery-mobile-1.2.0/js/widgets/forms/checkboxradio.js
@@ -94,7 +94,9 @@ $.widget( "mobile.checkboxradio", $.mobile.widget, {
 					event.stopPropagation();
 				}
 			},
-
+			vmousecancel: function ( event ) {
+				self.refresh();
+			},
 			vclick: function( event ) {
 				if ( input.is( ":disabled" ) ) {
 					event.preventDefault();
-- 
1.7.9.5

